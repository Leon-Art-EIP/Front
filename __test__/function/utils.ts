import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
