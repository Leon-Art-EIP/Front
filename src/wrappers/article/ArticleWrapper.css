.rich-text-content h1 {
  font-size: 2.5rem; /* Taille du texte pour les titres de niveau 1 */
  font-weight: bold; /* Poids de la police pour les titres de niveau 1 */
  margin-bottom: 1rem; /* Marge inférieure pour espacer les éléments */
}

.rich-text-content h2 {
  font-size: 2rem; /* Taille du texte pour les titres de niveau 2 */
  font-weight: bold; /* Poids de la police pour les titres de niveau 2 */
  margin-bottom: 0.75rem; /* Marge inférieure pour espacer les éléments */
}

.rich-text-content p {
  font-size: 1rem; /* Taille du texte pour les paragraphes */
  margin-bottom: 1rem; /* Marge inférieure pour espacer les paragraphes */
  line-height: 1.5; /* Hauteur de ligne pour améliorer la lisibilité */
}

.rich-text-content ul {
  list-style-type: disc; /* Type de puces pour les listes */
  padding-left: 1.5rem; /* Indentation des listes */
  margin-bottom: 1rem; /* Marge inférieure pour espacer les éléments */
}

.rich-text-content ol {
  list-style-type: decimal; /* Type de numérotation pour les listes */
  padding-left: 1.5rem; /* Indentation des listes */
  margin-bottom: 1rem; /* Marge inférieure pour espacer les éléments */
}
