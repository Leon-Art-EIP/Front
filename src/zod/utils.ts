export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
