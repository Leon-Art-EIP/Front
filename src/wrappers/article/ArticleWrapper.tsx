"use client";

import { useEffect, useState } from "react";
import { IPassingArt } from "../../interfaces/home/passingArt";
import { myFetch } from "../../tools/myFetch";
import { imageApi } from "../../tools/variables";
import "./ArticleWrapper.css"; // Assurez-vous d'importer votre fichier CSS

interface IArticleWrapperProps {
  id: string;
}

export default function ArticleWrapper(props: IArticleWrapperProps): JSX.Element {
  const [article, setArticle] = useState<IPassingArt>();

  useEffect(() => {
    async function fetchArticle() {
      const res = await myFetch({ route: `/api/article/${props.id}`, method: "GET" });
      if (res.ok) {
        const article = res.json as IPassingArt;
        setArticle(article);
        console.log(article.mainImage);
      }
    }
    fetchArticle();
  }, [props.id]);
  return (
    <div className="flex flex-col bg-background">
      <div className="h-64 w-screen bg-secondary">
        <img
          className="w-full h-full object-cover object-center"
          alt="articleBanner"
          // src={article?.mainImage}
          src={`${imageApi}/${article?.mainImage}`}
          // sizes={props.sizes}
          // style={props.style}
        />
      </div>
      <div className="flex w-full">
        <div className="w-40 p-4"></div>
        <div className="flex-1 bg-background p-4">
          <div className="mt-6 mb-10 text-4xl font-semibold truncate cursor-pointer text-tertiary text-center">
            {article?.title}
          </div>
          <div
            className="mt-4 text-tertiary rich-text-content"
            dangerouslySetInnerHTML={{ __html: article?.content || "" }}
          />
        </div>
        <div className="w-40 p-4"></div>
      </div>
    </div>
  );
}
