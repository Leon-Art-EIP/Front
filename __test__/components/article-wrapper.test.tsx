import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import ArticleWrapper from "../../src/wrappers/article/ArticleWrapper";

test("ArticleWrapper snapshot", () => {
  testSnapshot(<ArticleWrapper id="12" />);
});
