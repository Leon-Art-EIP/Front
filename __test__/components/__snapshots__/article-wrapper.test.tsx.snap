// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArticleWrapper snapshot 1`] = `
<div
  className="flex flex-col bg-background"
>
  <div
    className="h-64 w-screen bg-secondary"
  >
    <img
      alt="articleBanner"
      className="w-full h-full object-cover object-center"
      src="undefined/api/undefined"
    />
  </div>
  <div
    className="flex w-full"
  >
    <div
      className="w-40 p-4"
    />
    <div
      className="flex-1 bg-background p-4"
    >
      <div
        className="mt-6 mb-10 text-4xl font-semibold truncate cursor-pointer text-tertiary text-center"
      />
      <div
        className="mt-4 text-tertiary rich-text-content"
        dangerouslySetInnerHTML={
          {
            "__html": "",
          }
        }
      />
    </div>
    <div
      className="w-40 p-4"
    />
  </div>
</div>
`;
